todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/taskdetail/TaskDetailPresenter.java

```diff
@@ -1,151 +0,0 @@
-/*
- * Copyright 2016, The Android Open Source Project
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *      http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package com.example.android.architecture.blueprints.todoapp.taskdetail;
-
-import android.support.annotation.NonNull;
-import android.support.annotation.Nullable;
-
-import com.example.android.architecture.blueprints.todoapp.data.Task;
-import com.example.android.architecture.blueprints.todoapp.data.source.TasksRepository;
-import com.example.android.architecture.blueprints.todoapp.util.schedulers.BaseSchedulerProvider;
-import com.google.common.base.Strings;
-
-import io.reactivex.disposables.CompositeDisposable;
-
-import static com.google.common.base.Preconditions.checkNotNull;
-
-/**
- * Listens to user actions from the UI ({@link TaskDetailFragment}), retrieves the data and updates
- * the UI as required.
- */
-public class TaskDetailPresenter implements TaskDetailContract.Presenter {
-
-    @NonNull
-    private final TasksRepository mTasksRepository;
-
-    @NonNull
-    private final TaskDetailContract.View mTaskDetailView;
-
-    @NonNull
-    private final BaseSchedulerProvider mSchedulerProvider;
-
-    @Nullable
-    private String mTaskId;
-
-    @NonNull
-    private CompositeDisposable disposables;
-
-    public TaskDetailPresenter(@Nullable String taskId, @NonNull TasksRepository tasksRepository,
-                               @NonNull TaskDetailContract.View taskDetailView,
-                               @NonNull BaseSchedulerProvider schedulerProvider) {
-        this.mTaskId = taskId;
-        mTasksRepository = checkNotNull(tasksRepository, "tasksRepository cannot be null!");
-        mTaskDetailView = checkNotNull(taskDetailView, "taskDetailView cannot be null!");
-        mSchedulerProvider = checkNotNull(schedulerProvider, "schedulerProvider cannot be null");
-
-        disposables = new CompositeDisposable();
-        mTaskDetailView.setPresenter(this);
-    }
-
-    @Override
-    public void subscribe() {
-        openTask();
-    }
-
-    @Override
-    public void unsubscribe() {
-        disposables.clear();
-    }
-
-    private void openTask() {
-        if (Strings.isNullOrEmpty(mTaskId)) {
-            mTaskDetailView.showMissingTask();
-            return;
-        }
-
-        mTaskDetailView.setLoadingIndicator(true);
```

Comment:
I noticed you don't use this method in your code, is there a reason to it?
It no, you might need to actually do something in the `reducer > PopulateTask > IN_FLIGHT` case, i.e., set the viewState's `loading` field (non-existent right now) to true and set `setLoadingIndicator ` accordingly and setting it back to false in `FAILURE` on `SUCCESS`.

Reply:
That would be the last thing needed to be addressed before merging 👍

Reply:
done! thanks for the review @oldergod!